docs/dev-tools/zombienet/testing.md

```diff
@@ -0,0 +1,162 @@
+# Testing DSL
+
+Zombienet provides a Domain Specific Language (DSL) for writing tests. The DSL is designed to be human-readable and allows you to write tests using natural language expressions. Using this DSL, you can define assertions and tests against the spawned network. This way, users can evaluate different metrics, such as:
+
+- On chain storage
+- Metrics
+- Histograms
+- Logs
+- System events
+- Tracing
+- Custom api calls (through polkadot.js)
+- Commands 
+
+These abstractions are expressed by sentences defined in a natural language style. Therefore, each test line will be mapped to a test to run. Also, the test file (`*.zndsl`) includes pre-defined header fields used to define information about the suite, such as network configuration and credentials location.
+
+## Test file 
+
+### Name
+
+The test name in Zombienet is derived from the filename by removing any leading numeric characters before the first hyphen. For example, a file named “0001-zombienet-test.zndsl” will result in a test name of "zombienet-test" which will be displayed in the test runner’s report output.
+
+### Structure
+
+The test file is a text file with the extension `.zndsl`. It is divided into two parts: the header and the body. The header contains the network configuration and the credentials to use, while the body contains the tests to run.
+
+The header is defined by the following fields:
+
+| Key            | Type     | Description                                                                                     |
+| -------------- | -------- | ----------------------------------------------------------------------------------------------- |
+| `description`  | String (optional) | Long description of the test suite.                                                  |
+| `network`      | String   | Path to the network definition file, supported in both JSON and TOML formats.                   |
+| `creds`        | String   | Credentials file name or path to use (available only with Kubernetes provider). Looks in current directory or `$HOME/.kube/` if a filename is passed. |
+
+The body contains the tests to run. Each test is defined by a sentence in the DSL, which is mapped to a test to run. Each test line defines an assertion or a command to be executed against the spawned network.
+
+#### Assertions
+
+Assertions are defined by sentences in the DSL that evaluate different metrics, such as on-chain storage, metrics, histograms, logs, system events, tracing, and custom API calls. Each assertion is defined by a sentence in the DSL, which is mapped to a test to run.
+
+- `Well know functions` - already mapped test function
```

Comment:
I'm not sure if this is the right format or it should be:

- **Well know functions** - already mapped test function

Reply:
I find this section listing the assertions and examples to be difficult to read. This information may be better organized in a table perhaps?

Reply:
Yes, that makes sense. I'd re-structure this into a table tbh I was afraid of how it would look since it's  quite long, but let's give it a shot

Reply:
Not sure what it looked like before, but it feels weird to me as it is now on the live version. I'd have to play around with it a bit to provide a recommendation

Reply:
Thanks! Please feel free to provide it in [this new PR](https://github.com/papermoonio/polkadot-ecosystem-docs-draft/pull/8)